test/integration/oauth_ldap_test.go

```diff
@@ -0,0 +1,180 @@
+// +build integration,!no-etcd
+
+package integration
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/client"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+	testutil "github.com/openshift/origin/test/util"
+
+	"github.com/vjeantet/ldapserver"
+)
+
+func TestOAuthLDAP(t *testing.T) {
+	const (
+		providerName = "myldapprovider"
+
+		bindDN       = "o=company,uid=admin"
+		bindPassword = "admin-password"
+
+		myUserUID      = "myuser"
+		myUserName     = "My User, Jr."
+		myUserEmail    = "myuser@example.com"
+		myUserDN       = "o=company,myuid=" + myUserUID
+		myUserPassword = "myuser-password"
+	)
+
+	expectedSearchRequest := ldapserver.SearchRequest{
+		BaseObject:   []byte("o=company"),
+		Scope:        ldapserver.SearchRequestSingleLevel,
+		DerefAliases: 0,
+		SizeLimit:    2,
+		TimeLimit:    0,
+		TypesOnly:    false,
+		Attributes:   [][]byte{[]byte("cn"), []byte("dn"), []byte("mail"), []byte("myuid")},
+		Filter:       fmt.Sprintf("(&(myAttr=myValue)(myuid=%s))", myUserUID),
+	}
+
+	// Start LDAP server
+	ldapServer := testutil.NewTestLDAPServer()
+	ldapServer.SetPassword(bindDN, bindPassword)
+	ldapServer.Start("127.0.0.1:8389")
```

Comment:
Not crazy about this.  Why not a random port?  I would like to get to a point where we can run our integration tests in parallel.